evap/results/exporters.py

```diff
@@ -262,12 +288,24 @@ def write_questionnaire(self, questionnaire, evaluations_with_results, contribut
                 approval_count = 0
 
                 for grade_result in results[questionnaire.id]:
-                    if grade_result.question.id != question.id or not grade_result.has_answers:
+                    if grade_result.question.id != question.id:
+                        continue
+
+                    assert isinstance(grade_result, RatingResult)
+                    if not grade_result.has_answers:
                         continue
-                    values.append(grade_result.average * grade_result.count_sum)
-                    count_sum += grade_result.count_sum
-                    if grade_result.question.is_yes_no_question:
-                        approval_count += grade_result.approval_count
+
+                    # `filter_text_and_heading_questions` filters all text questions
+                    # `grade_result.question.id != question.id` filters the remaining HeadingResults because `question.is_heading_question` filters all heading questions
+                    # so here all `grade_result` from the OrderedDict values are only RatingResults with the same question as the current question which is not a heading question or a text question so it must be a rating question
+                    rating_result = cast(RatingResult, grade_result)
+                    assert hasattr(rating_result, "count_sum")
+                    assert hasattr(rating_result, "average")
```

Comment:
We have `assert isinstance(grade_result, RatingResult)` four lines above in 294, I don't understand the intention here. I think in the best case this PR doesn't have to touch any lines after the `continue` in 296.

Reply:
if grade_result.question.id != question.id: and if grade_result.question.is_yes_no_question: implicitly enforce the type so we assert to tell the typechecker